### PR TITLE
Fix: Use new Dart GH Action

### PR DESCRIPTION
**Pull request by ueman**
## :scroll: Description
Use new Dart GitHub Action and fix all Linter warnings in the examples

_#skip-changelog_

## :bulb: Motivation and Context
#334 and #374

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Comment by ueman** [comment]
#skip-changelog

**Comment by marandaneto** [comment]
@ueman it should be in the description, not PR's title, or as a comment, otherwise it does not work

**Comment by ueman** [comment]
> @ueman it should be in the description, not PR's title, or as a comment, otherwise it does not work

Dang it :D Thanks!

File: .github/workflows/dart.yml [hunk 1]
```diff
@@ -35,7 +33,7 @@ jobs:
             sdk: dev
 
     steps:
-        - uses: cedx/setup-dart@v2
+        - uses: dart-lang/setup-dart@v1
```
Comment (marandaneto) [inline .github/workflows/dart.yml:36]: its still using `release-channel` to config the channel, its different for this action

**Comment by marandaneto** [comment]
we still need to address https://github.com/getsentry/sentry-dart/issues/374 for the Flutter package, could be a different PR though, but I guess it already works for the Flutter package.

**Review by marandaneto — APPROVED** [review]

**Comment by ueman** [comment]
> we still need to address #374 for the Flutter package, could be a different PR though, but I guess it already works for the Flutter package.

We already build the example so I would say they're good as is:

https://github.com/getsentry/sentry-dart/blob/main/.github/workflows/flutter.yml#L62-L75

**Pull request merged**